src/context.rs

```diff
@@ -320,30 +321,51 @@ unsafe impl<'buf> PreallocatedContext<'buf> for AllPreallocated<'buf> {}
 unsafe impl<'buf> PreallocatedContext<'buf> for SignOnlyPreallocated<'buf> {}
 unsafe impl<'buf> PreallocatedContext<'buf> for VerifyOnlyPreallocated<'buf> {}
 
+/// A preallocated buffer, enforces the invariant that the buffer is big enough.
+#[allow(missing_debug_implementations)]
+pub struct PreallocatedBuffer<'buf>(&'buf [AlignedType]);
```

Comment:
This needs to be `mut` as it's unsound otherwise.

Also FYI I probably meant something else but can't remember what it was. I guess I meant `&mut AlignedType` which is unsound. We could also just use unsized type `pub struct PreallocatedBuffer([AligneType]);` which is more idiomatic.

In ideal world we would use an extern type but those aren't even stable.

Reply:
We should also probably make `AlignedType` contain `MaybeUninit` or something like that, as the C code might write padding bytes to it.

I think it wasn't stable (enough) back when I implemented this logic

Reply:
Yeah, it wasn't stable back then. Definitely should. I promoted this to https://github.com/rust-bitcoin/rust-secp256k1/issues/707 because I expect this old PR to get closed.